### ISSUE TITLE
investigate api data shape

### ISSUE THREAD

**Issue by nikrb** [chingu-voyage3/bears-21#18]
investigate data format for api comms.
- [ ] [normalizr](https://github.com/paularmstrong/normalizr) 
- [ ] [reselect](https://github.com/reactjs/reselect)
- [ ] [immutablejs](https://facebook.github.io/immutable-js/)

checkout [twilio](https://www.twilio.com/docs/api) to see how they do it.
useful link:
[redux dox](https://redux.js.org/docs/introduction/)
[normalizr with redux](https://tonyhb.gitbooks.io/redux-without-profanity/content/normalizer.html)

**Comment by nikrb** [comment]
changed feature to enhancement ... whilst we are here to learn, currently feeling well out of my depth.

**Comment by nikrb** [comment]
is it sufficient to have single global redux state for working and error, or does each 'module' require their own? Strictly speaking this is just frontend and not api related.

**Comment by vivekimsit** [comment]
As far as I know your app should have only single state tree.

**Comment by nikrb** [comment]
agreed. Does that tree have one global for working type situations and errors, or should each sub-tree have its own?
e.g.
```
{
isWorking: false,
hasErrored: { false, message:""},
user: { ... },
houses: { ... },
...
}
```
OR
```
{
user: { isWorking: false, hasErrored: {false, message: "", userData: {...}},
houses: { isWorking: false, hasErrored: {false, message: "", housesData: {...}},
}
...
```

**Comment by Mortuie** [comment]
@nikrb 2nd one. Otherwise how do you differentiate the errors :( Although the user will probably be only using one of the state trees.... Idk?
